.github/workflows/pre-commit.yaml

```diff
@@ -1,97 +1,77 @@
 name: pre-commit
 
 on:
-  push:
```

Comment:
This is better to show users on pull-requests and doesn't require any credential access so flipping to run on PRs

Reply:
Most of the other changes are just updates from the original workflow I created which now uses composite workflows